src/subdomains/supporting/support-issue/dto/create-support-message.dto.ts

```diff
@@ -0,0 +1,42 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import {
+  IsEnum,
+  IsNotEmpty,
+  IsNotEmptyObject,
+  IsOptional,
+  IsString,
+  ValidateIf,
+  ValidateNested,
+} from 'class-validator';
+import { EntityDto } from 'src/shared/dto/entity.dto';
+import { SupportIssue } from '../entities/support-issue.entity';
+import { SupportMessageAuthor } from '../entities/support-message.entity';
+
+export class CreateSupportMessageDto {
+  @ApiProperty({ type: EntityDto })
+  @IsNotEmptyObject()
+  @ValidateNested()
+  @Type(() => EntityDto)
+  supportIssue: SupportIssue;
+
+  @IsNotEmpty()
+  @IsEnum(SupportMessageAuthor)
+  author: SupportMessageAuthor;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  message?: string;
+
+  @ApiPropertyOptional({ description: 'Base64 encoded file' })
+  @IsOptional()
+  @IsString()
+  file?: string;
+
+  @ApiPropertyOptional({ description: 'Name of the file' })
+  @ValidateIf((l: CreateSupportMessageDto) => l.file != null)
+  @IsNotEmpty()
+  @IsString()
+  fileName?: string;
+}
```

Comment:
Maybe we should have a separate DTO for the reply, where the author can be specified? And maybe it should just be a string instead of enum?